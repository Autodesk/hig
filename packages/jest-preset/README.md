# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
