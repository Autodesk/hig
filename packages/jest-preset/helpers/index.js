/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
