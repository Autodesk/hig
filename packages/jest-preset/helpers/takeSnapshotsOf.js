const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
