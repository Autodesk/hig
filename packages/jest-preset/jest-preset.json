{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
