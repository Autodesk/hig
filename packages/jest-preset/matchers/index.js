require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
