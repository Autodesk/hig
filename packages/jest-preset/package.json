{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
