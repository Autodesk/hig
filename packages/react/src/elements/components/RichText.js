/* eslint-disable react/no-danger */
import React from "react";
import PropTypes from "prop-types";

function RichText(props) {
  return props.children ? (
    <div className="hig__rich-text">{props.children}</div>
  ) : (
    <div
      className="hig__rich-text"
      dangerouslySetInnerHTML={props.dangerouslySetInnerHTML}
    />
  );
}

RichText.propTypes = {
  children: PropTypes.node,
  dangerouslySetInnerHTML: PropTypes.shape({
    __html: PropTypes.string
  })
};

RichText.defaultProps = {
  children: null,
  dangerouslySetInnerHTML: null
};

RichText.__docgenInfo = {
  props: {
    children: {
      description: "react-generated markup to style with HIG typography rules"
    },
    dangerouslySetInnerHTML: {
      description:
        "HTML string to be rendered and styled with HIG typography rules"
    }
  }
};

export default RichText;
