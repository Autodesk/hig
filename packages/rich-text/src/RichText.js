import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import ThemeContext from "@hig/theme-context";

import stylesheet from "./stylesheet";

const RichText = props => (
  <ThemeContext.Consumer>
    {({ resolvedRoles }) => {
      const { children, dangerouslySetInnerHTML, ...otherProps } = props;
      const { className } = otherProps;
      const styles = stylesheet(props, resolvedRoles);
      return children ? (
        <div className={cx(css(styles.richText), className)}>{children}</div>
      ) : (
        <div
          className={cx(css(styles.richText), className)}
          dangerouslySetInnerHTML={dangerouslySetInnerHTML} // eslint-disable-line react/no-danger
        />
      );
    }}
  </ThemeContext.Consumer>
);

RichText.displayName = "RichText";

RichText.propTypes = {
  /**
   * React-generated markup to style with HIG typography rules
   */
  children: PropTypes.node,
  /**
   * HTML string to be rendered and styled with HIG typography rules
   */
  dangerouslySetInnerHTML: PropTypes.shape({
    __html: PropTypes.string
  }),
  /**
   * Adds custom/overriding styles
   */
  stylesheet: PropTypes.func
};

export default RichText;
