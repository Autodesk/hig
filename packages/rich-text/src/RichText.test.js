import React from "react";
import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";
import RichText from "./RichText";

describe("RichText", () => {
  takeSnapshotsOf(RichText, [{ desc: "renders", props: {} }]);
  takeSnapshotsOf(RichText, [
    {
      desc: "renders with children",
      props: { children: <h1>Happy little clouds</h1> },
    },
  ]);
  takeSnapshotsOf(RichText, [
    {
      desc: "renders with dangerouslySetInnerHTML",
      props: {
        dangerouslySetInnerHTML: {
          __html:
            "<p>Nice little fluffy clouds laying around in the sky being lazy.</p>",
        },
      },
    },
  ]);
  takeSnapshotsOf(RichText, [
    {
      desc: "renders with custom className",
      props: {
        className: "test-class",
        children: <h1>Happy little clouds</h1>,
      },
    },
  ]);
});
