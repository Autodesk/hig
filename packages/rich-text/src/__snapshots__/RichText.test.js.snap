// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText renders 1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 ul,
.emotion-0 ol {
  padding-left: 24px;
}

.emotion-0 ul li {
  list-style: none;
}

.emotion-0 ul li:before {
  content: '\\B7';
  vertical-align: middle;
  font-size: 20px;
  padding-right: 12px;
  margin-left: -14px;
}

.emotion-0 h1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.285714286;
  margin: 0;
  text-align: initial;
}

.emotion-0 h2 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.25;
  margin: 0;
  text-align: initial;
}

.emotion-0 h3 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

.emotion-0 a:hover {
  color: #006eaf;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #006eaf;
  text-decoration-color: #006eaf;
}

.emotion-0 a:focus {
  color: #006eaf;
  outline: solid 2px rgba(6,150,215,0.35);
}

.emotion-0 p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 h1 + p,
.emotion-0 h2 + p,
.emotion-0 h3 + p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 16px 0 12px 0;
  text-align: initial;
}

.emotion-0 b,
.emotion-0 strong {
  font-weight: 700;
}

<div
  className="emotion-0"
  dangerouslySetInnerHTML={undefined}
/>
`;

exports[`RichText renders with children 1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 ul,
.emotion-0 ol {
  padding-left: 24px;
}

.emotion-0 ul li {
  list-style: none;
}

.emotion-0 ul li:before {
  content: '\\B7';
  vertical-align: middle;
  font-size: 20px;
  padding-right: 12px;
  margin-left: -14px;
}

.emotion-0 h1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.285714286;
  margin: 0;
  text-align: initial;
}

.emotion-0 h2 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.25;
  margin: 0;
  text-align: initial;
}

.emotion-0 h3 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

.emotion-0 a:hover {
  color: #006eaf;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #006eaf;
  text-decoration-color: #006eaf;
}

.emotion-0 a:focus {
  color: #006eaf;
  outline: solid 2px rgba(6,150,215,0.35);
}

.emotion-0 p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 h1 + p,
.emotion-0 h2 + p,
.emotion-0 h3 + p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 16px 0 12px 0;
  text-align: initial;
}

.emotion-0 b,
.emotion-0 strong {
  font-weight: 700;
}

<div
  className="emotion-0"
>
  <h1>
    Happy little clouds
  </h1>
</div>
`;

exports[`RichText renders with custom className 1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 ul,
.emotion-0 ol {
  padding-left: 24px;
}

.emotion-0 ul li {
  list-style: none;
}

.emotion-0 ul li:before {
  content: '\\B7';
  vertical-align: middle;
  font-size: 20px;
  padding-right: 12px;
  margin-left: -14px;
}

.emotion-0 h1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.285714286;
  margin: 0;
  text-align: initial;
}

.emotion-0 h2 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.25;
  margin: 0;
  text-align: initial;
}

.emotion-0 h3 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

.emotion-0 a:hover {
  color: #006eaf;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #006eaf;
  text-decoration-color: #006eaf;
}

.emotion-0 a:focus {
  color: #006eaf;
  outline: solid 2px rgba(6,150,215,0.35);
}

.emotion-0 p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 h1 + p,
.emotion-0 h2 + p,
.emotion-0 h3 + p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 16px 0 12px 0;
  text-align: initial;
}

.emotion-0 b,
.emotion-0 strong {
  font-weight: 700;
}

<div
  className="emotion-0 test-class"
>
  <h1>
    Happy little clouds
  </h1>
</div>
`;

exports[`RichText renders with dangerouslySetInnerHTML 1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 ul,
.emotion-0 ol {
  padding-left: 24px;
}

.emotion-0 ul li {
  list-style: none;
}

.emotion-0 ul li:before {
  content: '\\B7';
  vertical-align: middle;
  font-size: 20px;
  padding-right: 12px;
  margin-left: -14px;
}

.emotion-0 h1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.285714286;
  margin: 0;
  text-align: initial;
}

.emotion-0 h2 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.25;
  margin: 0;
  text-align: initial;
}

.emotion-0 h3 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

.emotion-0 a:hover {
  color: #006eaf;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #006eaf;
  text-decoration-color: #006eaf;
}

.emotion-0 a:focus {
  color: #006eaf;
  outline: solid 2px rgba(6,150,215,0.35);
}

.emotion-0 p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0 0 12px 0;
  text-align: initial;
}

.emotion-0 h1 + p,
.emotion-0 h2 + p,
.emotion-0 h3 + p {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 16px 0 12px 0;
  text-align: initial;
}

.emotion-0 b,
.emotion-0 strong {
  font-weight: 700;
}

<div
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Nice little fluffy clouds laying around in the sky being lazy.</p>",
    }
  }
/>
`;
